src/main/java/com/flightstats/hub/dao/ChannelService.java

```diff
@@ -57,44 +59,51 @@
 
     private final static Logger logger = LoggerFactory.getLogger(ChannelService.class);
     private static final int DIR_COUNT_LIMIT = HubProperties.getProperty("app.directionCountLimit", 10000);
-    @Inject
     private ContentService contentService;
-    @Inject
-    @Named("ChannelConfig")
     private Dao<ChannelConfig> channelConfigDao;
-    @Inject
-    private ChannelValidator channelValidator;
-    @Inject
-    private ReplicationManager replicationManager;
-    @Inject
+    private Provider<ChannelValidator> channelValidator;
+    private WatchManager watchManager;
     private LastContentPath lastContentPath;
-    @Inject
     private InFlightService inFlightService;
-    @Inject
     private TimeService timeService;
-    @Inject
     private StatsdReporter statsdReporter;
 
+    @Inject
+    public ChannelService(
+            ContentService contentService,
+            @Named("ChannelConfig") Dao<ChannelConfig> channelConfigDao,
+            Provider<ChannelValidator> channelValidator,
+            WatchManager watchManager,
+            LastContentPath lastContentPath,
+            InFlightService inFlightService,
+            TimeService timeService,
+            StatsdReporter statsdReporter) {
+        this.contentService = contentService;
+        this.channelConfigDao = channelConfigDao;
+        this.channelValidator = channelValidator;
+        this.watchManager = watchManager;
+        this.lastContentPath = lastContentPath;
+        this.inFlightService = inFlightService;
+        this.timeService = timeService;
+        this.statsdReporter = statsdReporter;
+    }
+
     public boolean channelExists(String channelName) {
         return channelConfigDao.exists(channelName);
     }
 
     public ChannelConfig createChannel(ChannelConfig configuration) {
         logger.info("create channel {}", configuration);
-        channelValidator.validate(configuration, null, false);
+        channelValidator.get().validate(configuration, null, false);
         channelConfigDao.upsert(configuration);
         notify(configuration, null);
         TagWebhook.updateTagWebhooksDueToChannelConfigChange(configuration);
         return configuration;
     }
 
     private void notify(ChannelConfig newConfig, ChannelConfig oldConfig) {
-        if (newConfig.isReplicating()) {
-            replicationManager.notifyWatchers();
-        } else if (oldConfig != null) {
-            if (oldConfig.isReplicating()) {
-                replicationManager.notifyWatchers();
-            }
+        if (newConfig.isReplicating() || (oldConfig != null && oldConfig.isReplicating())) {
```

Comment:
Your way is easy to understand Chris. As much as I hate helper classes/methods, here there is not much choice for now. Each class does too many things. Hopefully will move towards SRP eventually.

Reply:
Heh, I use helper methods pretty frequently :), though abuse of it can just makes things worse.

If I'm looking at a few lines of code, or a complicated line of code, and find myself trying to describe the aggregate behavior then I'll ask myself "do these represent some concept?". If the answer is yes, then I'll often extract a helper method (much more rarely, a helper class) with that concept as it's name.